controller/controller.go

```diff
@@ -239,78 +233,68 @@ func (c *Controller) deleteFunc(ctx context.Context, obj interface{}) {
 	}
 }
 
-// ProcessEvents takes the event channels created by the operatorkit informer
-// and executes the controller's event functions accordingly.
-func (c *Controller) ProcessEvents(ctx context.Context, deleteChan chan watch.Event, updateChan chan watch.Event, errChan chan error) error {
-	loop := -1
+// Reconcile implements the reconciler given to the controller-runtime
+// controller. Reconcile never returns any error as we deal with them in
+// operatorkit internally.
+func (c *Controller) Reconcile(req reconcile.Request) (reconcile.Result, error) {
+	ctx := context.Background()
 
-	for {
-		loop++
+	res, err := c.reconcile(ctx, req)
+	if err != nil {
+		c.logger.LogCtx(ctx, "level", "error", "message", "failed to reconcile", "stack", microerror.Stack(err))
+		return reconcile.Result{}, nil
+	}
 
-		// Set loop specific logger context.
-		{
-			ctx = setLoggerCtxValue(ctx, loggerKeyLoop, strconv.Itoa(loop))
-		}
+	return res, nil
+}
 
-		select {
-		case e := <-deleteChan:
-			event := "delete"
+func (c *Controller) reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
```

Comment:
The method order in this file is a fucking mess. I want to fix that in a follow up PR. Just did not want to blow it here.